### DISCUSSION TITLE
0.12.0-beta.1 - ENDED

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#201] [Announcements]
This is not a normal version. It's a beta test version and can be unstable. [See these instructions, if you are willing to participate in beta testing](https://github.com/Taitava/obsidian-shellcommands/discussions/98). Do not use any important vault for testing, because you are not able to convert your data.json settings file back to an earlier version!

Main things for testing:
 - Prompts for asking values from a user, #17
 - Custom variables, #146
 - Default values for variables, #190
 - Hotkeys for output channel: Ask after execution, #145

Documentation:
- Prompts: https://publish.obsidian.md/shellcommands/Preactions/Prompts
- Custom variables: https://publish.obsidian.md/shellcommands/Variables/Custom+variables
- Default values for variables: https://publish.obsidian.md/shellcommands/Variables/Default+values

Because prompts and custom variables are quite big features, I decided to record a 22 minutes long video showing beta testers how prompts and custom variables can be created and used. It's not meant to be an actual documentation video (the video is temporary), so it's only visible when you know the link, which is here: https://youtu.be/qjGNP7F2ozY

<hr /><em>This discussion was created from the release <a href='https://github.com/Taitava/obsidian-shellcommands/releases/tag/0.12.0-beta.1'>0.12.0-beta.1</a>.</em>

**Comment by FelipeRearden** [comment]
Hello @Taitava !!!!!!!!!!!

The new beta look amazing. We a have a new set of great features !!!!

If don’t mind, just give a few days to start my tests routines. 

We have a lot of ground to cover and I want to do everything with max attention. 

Have a great day !!!!!!!!!!!!!!!!!!!!!!!!!!!

**Reply by Taitava** [reply]
Thank you @FelipeRearden! ❤️ Sure, take your time! I don't have schedules. I don't schedule when I get beta tests ready (= no prior information that a beta is coming on day x), so I also won't expect that people would happen to have time to do beta testing at the moment when I happen to release a beta. Test whenever you have time! 🙂 Be it days or weeks, there is no deadlines, things get completed eventually👍. Sometimes there's a pause and that's just a good thing. 🌞

Have a great day, too! ❤️

**Reply by FelipeRearden** [reply]
You are super @Taitava !!!! ❤️

Thank you so much for understanding. As you know I care about SC and about you very much. 💙 💙 💙 💙 💙 💙 💙 💙 

And I wanna save some time in the next days to help in testing the beta. 

I keep you posted !!!!

I wish you a fantastic Monday !!!!!!!!!!

**Reply by Taitava** [reply]
Thank you, you are super, too! ❤️ You've helped this project a lot and it means so much to me!

Have a great Tuesday! 😎

**Comment by imed-ghomari** [comment]
I tried the new prompt, it's amazing. Just one nitpick, When writing in the last field, to execute the prompt, I have to press `tab` then `enter`. It would be better if pressing `enter` in the last field would execute the prompt directly. 
Thank you again for this new feature.

Edit: make `enter` in any field execute the prompt. This is better because some fields may be not required, so going down to the last field and pressing `enter` would be counterproductive.

**Reply by Taitava** [reply]
Thank you for your feedback!

The `enter` thing is a really valuable idea! I'm a bit stuck with planning it. Generally, all of the following modals in the plugin could benefit from the ability to hit the `enter` key to approve the modal:
- The prompt modal, like you suggested
- Deletion modal (to delete a shell command, a prompt, a prompt field, or a custom variable)
- Perhaps the _Ask after execution_ output modal, but only when the focus is not in a textarea field, where `enter` inserts a line break.
- Confirmation modal (when a shell command is defined to ask a confirmation before execution)

The problem I'm thinking about is hitting `enter` accidentally. For example, when using Obsidian's command palette to select a shell command to be executed, then hit `enter` to select that command. If the shell command requires a confirmation before execution, a confirmation modal opens up. If `enter` is accidentally hit twice in the command palette, then the second hit will approve the confirmation, and accidentally execute the shell command.

So, on one hand, if I implement the `enter` key ability to approve modals, I'd like to do it for all of the above mentioned modals in order to keep the user experience consistent. On the other hand, I'm a bit caucious about miss hits.

One way to solve this would be to introduce a setting which allows turning the `enter` key ability off if it causes accidents.

**Reply by imed-ghomari** [reply]
> One way to solve this would be to introduce a setting which allows turning the enter key ability off if it causes accidents.

Yes, a toggle like that would be best for people who are prone to this type of accident. As for others, they could just leave it on.

**Reply by Taitava** [reply]
Good. I'll implement this in some later version, either `0.13.0` or `0.14.0`.

**Reply by Taitava** [reply]
There's now an issue for tracking the process of the enter key feature: #216

**Reply by Taitava** [reply]
@imeed166 it's now released in version `0.13.0`. 🙂

**Reply by imed-ghomari** [reply]
It works great, thank you.

**Reply by Taitava** [reply]
Thanks! 🙂

**Comment by Taitava** [comment]
# 0.12.0 is now released

The beta test is ended.
